From: Andor Daam <andor.daam@googlemail.com>
Date: Mon, 5 Dec 2011 10:10:37 +0100
Subject: [009/106] Staging: line6/midi.c: Fixed call of obsolete function
 strict_strtoul

commit 251c3948808046c39359658fa2faffff45b825d3 upstream.

The obsolete function strict_strtoul should be replaced by the kstrto*
functions.

In this context kstrtou16 should be use, as midi_mask_receive is only
used as unsigned short. All corresponding datatypes were adapted
accordingly.

Signed-off-by: Andor Daam <andor.daam@googlemail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/line6/midi.c |    4 ++--
 drivers/staging/line6/midi.h |    2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

Index: linux-3.2.46/drivers/staging/line6/midi.c
===================================================================
--- linux-3.2.46.orig/drivers/staging/line6/midi.c	2013-07-26 19:28:07.000000000 +0000
+++ linux-3.2.46/drivers/staging/line6/midi.c	2013-07-26 19:28:08.000000000 +0000
@@ -339,10 +339,10 @@
 {
 	struct usb_interface *interface = to_usb_interface(dev);
 	struct usb_line6 *line6 = usb_get_intfdata(interface);
-	unsigned long value;
+	unsigned short value;
 	int ret;
 
-	ret = strict_strtoul(buf, 10, &value);
+	ret = kstrtou16(buf, 10, &value);
 	if (ret)
 		return ret;
 
Index: linux-3.2.46/drivers/staging/line6/midi.h
===================================================================
--- linux-3.2.46.orig/drivers/staging/line6/midi.h	2013-07-26 19:28:07.000000000 +0000
+++ linux-3.2.46/drivers/staging/line6/midi.h	2013-07-26 19:28:08.000000000 +0000
@@ -62,7 +62,7 @@
 	/**
 		 Bit mask for input MIDI channels.
 	*/
-	int midi_mask_receive;
+	unsigned short midi_mask_receive;
 
 	/**
 		 Buffer for incoming MIDI stream.
